backend/runner/plancheck/statement_report_executor.go

```diff
@@ -410,42 +405,34 @@ func reportForMySQL(ctx context.Context, sqlDB *sql.DB, engine storepb.Engine, d
 	var totalAffectedRows int64
 	var changedResources []base.SchemaResource
 
-	p := tidbparser.New()
-	p.EnableWindowFunc(true)
-
 	for _, stmt := range singleSQLs {
-		if stmt.Empty {
-			continue
-		}
-		if tidb.IsTiDBUnsupportDDLStmt(stmt.Text) {
+		if stmt.Empty || stmt.Text == "" {
 			continue
 		}
-		root, _, err := p.Parse(stmt.Text, charset, collation)
+
+		stmts, err := mysqlparser.ParseMySQL(statement)
 		if err != nil {
 			slog.Error("failed to parse statement", slog.String("statement", stmt.Text), log.BBError(err))
 			continue
 		}
 
-		if len(root) != 1 {
+		if len(stmts) != 1 {
 			slog.Debug("failed to parse statement, expect to get one node from parser", slog.String("statement", stmt.Text))
 			continue
 		}
-		sqlType, resources := getStatementTypeFromTidbAstNode(strings.ToLower(databaseName), root[0])
+
+		sqlType := mysqlparser.GetStatementType(stmts[0])
 		sqlTypeSet[sqlType] = struct{}{}
 		if !isDML(sqlType) {
-			if engine != storepb.Engine_TIDB {
-				resources, err := base.ExtractChangedResources(storepb.Engine_MYSQL, databaseName, "" /* currentSchema */, stmt.Text)
-				if err != nil {
-					slog.Error("failed to get statement changed resources", log.BBError(err))
-				} else {
-					changedResources = append(changedResources, resources...)
-				}
+			resources, err := base.ExtractChangedResources(storepb.Engine_MYSQL, databaseName, "" /* currentSchema */, stmt.Text)
+			if err != nil {
+				slog.Error("failed to extract changed resources", slog.String("statement", stmt.Text), log.BBError(err))
 			} else {
 				changedResources = append(changedResources, resources...)
 			}
 		}
 
-		affectedRows, err := getAffectedRowsForMysql(ctx, engine, sqlDB, dbMetadata, root[0])
+		affectedRows, err := getAffectedRowsForMySQL(ctx, engine, sqlDB, dbMetadata, stmts[0])
```

Comment:
To put it simply, use `base.GetAffectedRows(ctx, engine, stmt, buildGetRowsCountByQuery(sqlDB, engine), buildGetTableDataSizeFunc(metadata))` here.
Otherwise, LGTM.